bayesian/manifest_models.py

```diff
@@ -0,0 +1,248 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""Model to parse manifest files."""
+
+from lxml import etree
+from lxml import objectify
+
+
+class MavenPom:
+    """Model for Maven POM (Project Object Model)."""
+
+    def __init__(self, document=None):
+        """Initialize constructor for MavenPom class.
+
+        :document: Parse the content of the file.
+        :returns: None
+
+        """
+        if not document:
+            raise ValueError("No content is provided for parsing")
+
+        self.document = document.strip()
+        if not isinstance(self.document, (bytes, bytearray)):
+            self.document = self.document.encode()
+
+        try:
+            self.root = objectify.fromstring(self.document)
+            if getattr(self.root, 'dependencies', None) is None:
+                # create a dependencies element if doesn't exist
+                _prev = getattr(self.root, 'dependencyManagement', None)\
+                    or getattr(self.root, 'properties', None)\
+                    or getattr(self.root, 'name', None)
+                if _prev is not None:
+                    _prev.addnext(objectify.Element('dependencies'))
+                else:
+                    self.root.dependencies = objectify.ObjectifiedElement()
+                self.root = self._reload(self.root)
+
+        except etree.XMLSyntaxError as exc:
+            print("Unable to parse xml document:\n {}".format(str(exc)))
```

Comment:
should not you raise exception there as well? Because the `self.root` wont be initialized properly in case of exception.